HereBeDragons-Pins-1.0.lua

```diff
@@ -276,7 +276,7 @@ local function UpdateMinimapZoom()
 end
 
 local function PositionWorldMapIcon(icon, data, currentMapID, currentMapFloor)
-    local x, y = HBD:GetZoneCoordinatesFromWorld(data.x, data.y, currentMapID, currentMapFloor)
+    local x, y = HBD:GetZoneCoordinatesFromWorld(data.x, data.y, currentMapID, (currentMapID ~= 0 and currentMapFloor or data.instanceID))
```

Comment:
This condition should be flipped around, so that a "nil" floor doesn't fall back to sending it the instance id, as that would break things (Lua and/or constructs are not true ternary operations)

ie. "currentMapID == 0 and data.instance or currentMapFloor"